### DISCUSSION TITLE
Does the Python package install the CLI ?

### DISCUSSION THREAD

**Discussion by ReHoss** [python-jsonschema/jsonschema#970] [Q&A] [answered]
Hi everyone,

Sorry if my question is naive.

I installed the python package `jsonschema` in a new conda environment through `conda install jsonschema`. I feel like this installed a binary in `/usr/bin/jsonschema` and added it to the bash `PATH`.

Am I right ? If so, I did not know a conda package could modify `PATH` and `/usr/bin`. It's fine but have you some specific references/explainations on this behavior for conda/pip in order to better understand what I do with my computer ?

Thank you,

**Comment by Julian** [comment, marked as answer]
The answer to the question in the title is certainly yes :)

I don't know much about how conda behaves on various systems though unfortunately. I do know by default that it installs into a particular conda env, and the directory for binaries for an env is unlikely to be `/usr/bin`, but I don't know whether any Linux distributions' conda package have the base conda environment be that, nor do I know whether they make installing things into it not require root. It seems unlikely (especially non-root), but remotely possible.

pip on various distros, if you install it to the system python via e.g. `sudo apt install python3-pip` on some Debianish system, will indeed then have `sudo pip install foo` install its binaries in `/usr/bin`, but no one should run `sudo pip` (and probably one should mostly not use the `python3-pip` package).

Beyond that as I say I'm not sure what conda does on various distros but you could try it out in a container and see perhaps, or someone else may know.

**Reply by Julian** [reply]
FWIW /usr/bin is almost certainly already on your PATH by the way, so if it did install things in there, the PATH manipulation isn't necessary.
